src/htmx.js

```diff
@@ -2313,7 +2313,7 @@ var htmx = (function() {
       return false
     }
     if (evt.type === 'submit' || evt.type === 'click') {
-      if (elt.tagName === 'FORM') {
+      if (elt.tagName === 'FORM' && asElement(evt.target)?.tagName === 'FORM') {
```

Comment:
I'm wondering ; couldn't we simply check `if (asElement(evt.target) || elt).tagName === 'FORM'` in this case?

So that, if defined, `evt.target` takes precedence, otherwise we keep the existing behavior. You would probably not even need to add the `target` property to mock the events of the test suite in this case? As `elt` is the fallback

Btw, should we use the same technique for the other checks in this same function? Anchors for ex. Maybe we should simply change to`const elt = asElement(evt.target) || asElement(node)` at the start of the function and not need to modify anything else inside?
Just some thoughts, completely untested, let me know!

Reply:
@Telroshan the problem is, the event target is an [EventTarget](https://developer.mozilla.org/en-US/docs/Web/API/EventTarget) and an EventTarget can be a Dom element but doesn't have to be. So the property tagname can be undefined. (It was my first approach but the type checker complaint). That's why I chose this way to ensure we only check properties which exist and don't run into any errors when trying to access undefined properties

Reply:
> an EventTarget can be a Dom element but doesn't have to be. So the property tagname can be undefined.

Hence my suggestion @pfiadDi : `const elt = asElement(evt.target) || asElement(node)`
Unless I'm mistaked (could totally be, didn't test this 😆 ), this will get the `evt.target` if it's a DOM element, but if it's not an element, will fallback to `node`.
Then we have the already present `if` statement that returns early, in case neither `evt.target` nor `node` were elements.
I believe you shouldn't get any type checker complaint with that approach as we cast to `Element` thanks to `asElement`?